scripts/upgrade

```diff
@@ -1,25 +1,27 @@
 #!/bin/bash
 
+source _common.sh
 source /usr/share/yunohost/helpers
 
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
 
-if [[ "${redirect_type:-}" == *"proxy"* ]]
-then
+if [[ "${redirect_type:-}" == *"proxy"* ]]; then
     redirect_type="reverseproxy"
 else
     redirect_type="redirect"
 fi
 ynh_app_setting_set $app 'redirect_type' $redirect_type
 
-if [[ -z "${target:-}" ]] && [[ -n "${redirect_path:-}" ]]
-then
+if [[ -z "${target:-}" ]] && [[ -n "${redirect_path:-}" ]]; then
     target="$redirect_path"
-    ynh_app_setting_set $app target "$redirect_path"
+    ynh_app_setting_delete --app=$app --key=redirect_path
+    ynh_app_setting_set --app=$app --key=target --value="$redirect_path"
 fi
 
+_validate_redirect_uri
```

Comment:
I dunno what to do with this ... if somebody somehow bypassed the constrain on the URL, do we really want to miserably crash the upgrade ... the regex is meant to check to prevent "unencrypted remote destination with reverse proxy mode" which doesnt mean that it cant technically work, just that it's insecure...

The check should be enforced when we set the variable (during install and config panel) but no need to check during upgrade ...